extension/src/experiments/model/status/collect.ts

```diff
@@ -169,3 +211,100 @@ export const collectSelected = (
     availableColors
   )
 }
+
+export const collectStartedRunningExperiments = (
+  previouslyRunning: { id: string; executor: string }[],
+  nowRunning: { id: string; executor: string }[]
+): Set<string> => {
+  const acc = new Set<string>()
+
+  for (const { id: runningId, executor } of nowRunning) {
+    if (previouslyRunning.some(({ id }) => id === runningId)) {
+      continue
+    }
+    acc.add(executor === 'workspace' ? 'workspace' : runningId)
+  }
+
+  return acc
+}
+
+const getMostRecentExperiment = (
+  experiments: Experiment[],
+  coloredStatus: ColoredStatus
+): Experiment | undefined =>
+  experiments
+    .filter(({ id }) => coloredStatus[id] === undefined)
+    .sort(({ Created: aCreated }, { Created: bCreated }) => {
+      if (!aCreated) {
+        return 1
+      }
+      if (!bCreated) {
+        return -1
+      }
+      return bCreated.localeCompare(aCreated)
+    })
+    .slice(0, 1)[0]
+
+const collectFinishedWorkspaceExperiment = (
+  acc: { [id: string]: string },
+  mostRecentExperiment: Experiment | undefined
+): void => {
+  const newId = mostRecentExperiment?.id
+
+  if (!newId) {
+    return
+  }
+
+  for (const [id, previousId] of Object.entries(acc)) {
+    if (previousId === 'workspace') {
+      delete acc[id]
+    }
+  }
+  acc[newId] = 'workspace'
+}
+
+const isStillRunning = (
+  stillExecutingInWorkspace: boolean,
+  previouslyRunningId: string,
+  previouslyRunningExecutor: string,
+  stillRunning: { id: string; executor: string }[]
+): boolean =>
+  (stillExecutingInWorkspace && previouslyRunningExecutor === 'workspace') ||
+  stillRunning.some(({ id }) => id === previouslyRunningId)
+
+export const collectFinishedRunningExperiments = (
+  acc: { [id: string]: string },
+  experiments: Experiment[],
+  previouslyRunning: { id: string; executor: string }[],
+  stillRunning: { id: string; executor: string }[],
+  coloredStatus: ColoredStatus
+): { [id: string]: string } => {
+  const stillExecutingInWorkspace = stillRunning.some(
+    ({ executor }) => executor === 'workspace'
+  )
+  for (const {
+    id: previouslyRunningId,
+    executor: previouslyRunningExecutor
+  } of previouslyRunning) {
+    if (
+      isStillRunning(
+        stillExecutingInWorkspace,
+        previouslyRunningId,
+        previouslyRunningExecutor,
+        stillRunning
+      )
+    ) {
+      continue
+    }
+
+    if (previouslyRunningExecutor === 'workspace') {
+      collectFinishedWorkspaceExperiment(
+        acc,
+        getMostRecentExperiment(experiments, coloredStatus)
```

Comment:
[F] This is an assumption that the most recently created "unseen" experiment is the new one.